src/inferenceresponse.cpp

```diff
@@ -39,6 +39,15 @@ Status InferenceResponse::addOutput(const std::string& name, DataType datatype,
     auto [it, emplaced] = outputs.emplace(name, InferenceTensor{datatype, shape, dimCount});
     return emplaced ? StatusCode::OK : StatusCode::DOUBLE_TENSOR_INSERT;
 }
+Status InferenceResponse::addOutput(const std::string& name) {
+    // TODO insert tensor with wrong shape/datatype/name/dimcount
+    // TODO reuse infer response/request
+    auto [it, emplaced] = outputs.emplace(name, InferenceTensor{});
+    return emplaced ? StatusCode::OK : StatusCode::DOUBLE_TENSOR_INSERT;
+}
+bool InferenceResponse::hasOutput(const std::string& name) const {
```

Comment:
Redundant -> you can call getOutput & check i you get nullptr. In case you find the output you can also use it straightaway instead of searching for it second time.

Reply:
done